### PR TITLE
Reorder methods in `Queue`, `Ref`, `RefM`, `Promise`, `Chunk`, `Exit`, `Cause` and `Semaphore`

### PR DESCRIPTION
**Pull request by regiskuckaertz**

### CODE REVIEW

File: core/shared/src/main/scala/scalaz/zio/Exit.scala [hunk 1]
```diff
@@ -24,39 +24,39 @@ sealed trait Exit[+E, +A] extends Product with Serializable { self =>
...
   /**
-   * Converts the `Exit` to an `Either[Throwable, A]`, by wrapping the
-   * cause in `FiberFailure` (if the result is failed).
+   * Sequentially zips the this result with the specified result discarding the second element of the tuple or else returns the failed `Cause[E1]`
    */
-  final def toEither: Either[Throwable, A] = self match {
-    case Success(value) => Right(value)
-    case Failure(cause) => Left(FiberFailure(cause))
-  }
+  final def <*[E1 >: E, B](that: Exit[E1, B]): Exit[E1, A] = zipWith(that)((_, _), _ ++ _).map(_._1)
```
Comment (LGLO) [inline core/shared/src/main/scala/scalaz/zio/Exit.scala]: In alphanum names when one method name is a prefix of other methods name, it's placed before.
This is prefix of `<*>`.

Comment (regiskuckaertz) [reply core/shared/src/main/scala/scalaz/zio/Exit.scala]: Thanks, you're right. I've reordered (thanks scaladoc for doing all the work 😅 ).

**Comment by regiskuckaertz** [comment]
@LGLO have also done `ZQueue` if you don't mind a final look.

**Comment by jdegoes** [comment]
@regiskuckaertz Awesome work!

**Pull request merged**